src/software/ai/hl/stp/tactic/chip_tactic_test.cpp

```diff
@@ -2,15 +2,274 @@
+    world =
+        ::Test::TestUtil::setBallPosition(world, Point(0, 0), Timestamp::fromSeconds(0));
+
+    Robot robot = Robot(0, Point(-1.3, 2), Vector(0, 0), Angle::zero(),
+                        AngularVelocity::zero(), Timestamp::fromSeconds(0));
+
+    ChipTactic tactic = ChipTactic(world.ball(), true);
+
+    tactic.updateRobot(robot);
+    tactic.updateControlParams(Point(0, 0), Point(-1, 1), 2.0);
+    auto action_ptr = tactic.getNextAction();
+
+    // Check an action was returned (the pointer is not null)
+    EXPECT_TRUE(action_ptr);
+
+    auto chip_action = std::dynamic_pointer_cast<ChipAction>(action_ptr);
+    ASSERT_NE(chip_action, nullptr);
+    EXPECT_EQ(Point(0, 0), chip_action->getChipOrigin());
+    EXPECT_EQ(Angle::fromDegrees(135.0), chip_action->getChipDirection());
+    EXPECT_EQ(2.0, chip_action->getChipDistanceMeters());
+}
+
+TEST(ChipTacticTest, robot_behind_ball_chipping_towards_negative_x_negative_y)
+{
+    World world = ::Test::TestUtil::createBlankTestingWorld();
+    world =
+        ::Test::TestUtil::setBallPosition(world, Point(0, 0), Timestamp::fromSeconds(0));
+
+    Robot robot = Robot(0, Point(-0.5, 1.3), Vector(0, 0), Angle::zero(),
+                        AngularVelocity::zero(), Timestamp::fromSeconds(0));
+
+    ChipTactic tactic = ChipTactic(world.ball(), true);
+
+    tactic.updateRobot(robot);
+    tactic.updateControlParams(Point(0, 0), Point(-1, -1), 2.0);
+    auto action_ptr = tactic.getNextAction();
+
+    // Check an action was returned (the pointer is not null)
+    EXPECT_TRUE(action_ptr);
+
+    auto chip_action = std::dynamic_pointer_cast<ChipAction>(action_ptr);
+    ASSERT_NE(chip_action, nullptr);
+    EXPECT_EQ(Point(0, 0), chip_action->getChipOrigin());
+    EXPECT_EQ(Angle::fromDegrees(225.0), chip_action->getChipDirection());
+    EXPECT_EQ(2.0, chip_action->getChipDistanceMeters());
+}
+
+TEST(ChipTacticTest, robot_behind_ball_chipping_towards_positive_x_negative_y)
+{
+    World world = ::Test::TestUtil::createBlankTestingWorld();
+    world =
+        ::Test::TestUtil::setBallPosition(world, Point(0, 0), Timestamp::fromSeconds(0));
+
+    Robot robot = Robot(0, Point(-1.2, 2.1), Vector(0, 0), Angle::zero(),
+                        AngularVelocity::zero(), Timestamp::fromSeconds(0));
+
+    ChipTactic tactic = ChipTactic(world.ball(), true);
+
+    tactic.updateRobot(robot);
+    tactic.updateControlParams(Point(0, 0), Point(1, -1), 2.0);
+    auto action_ptr = tactic.getNextAction();
+
+    // Check an action was returned (the pointer is not null)
+    EXPECT_TRUE(action_ptr);
+
+    auto chip_action = std::dynamic_pointer_cast<ChipAction>(action_ptr);
+    ASSERT_NE(chip_action, nullptr);
+    EXPECT_EQ(Point(0, 0), chip_action->getChipOrigin());
+    EXPECT_EQ(Angle::fromDegrees(315.0), chip_action->getChipDirection());
+    EXPECT_EQ(2.0, chip_action->getChipDistanceMeters());
+}
+
+TEST(ChipTacticTest, robot_not_behind_ball_chipping_towards_positive_x_positive_y)
```

Comment:
Could we be a bit more specific here and say "besides" or "in front of" the ball? I'm also not super clear on what "behind" actually means here. Is it related to the direction of kicking?

Reply:
Was following tests from `KickTactic` where they did "behind_ball" and "not_behind_ball". I believe it refers to position of robot relative to ball when kicking.

Reply:
On second consideration, seems fine with me.